### ISSUE TITLE
HTMLCollection property names shouldn't be enumerable

### ISSUE THREAD

**Issue by frewsxcv** [servo/servo#9315]
```html
<form id="form1" action=""></form>

<script>
var result = [];
for (var p in document.forms) {
  result.push(p);
}
console.log(result);
</script>
```

Expected output:

```
0,item,namedItem,length
```

Actual output:

```
0,form1,item,namedItem,length
```

I think this is blocked on #4043

**Comment by nox** [comment]
#9067

**Comment by frewsxcv** [comment]
[Current failing test](https://github.com/servo/servo/blob/d28fb42201dd743c470fcc1f43bf2e13f1c8cddc/tests/wpt/web-platform-tests/html/dom/documents/dom-tree-accessors/document.forms.html#L51-L57)

**Comment by jdm** [comment]
Fixed by #9067.
